api/routers/chat.py

```diff
@@ -80,3 +80,51 @@ async def get_thought(conversation_id: str, message_id: str, user_id: str):
     )
     # In practice, there should only be one thought per message
     return {"thought": thought.items[0].content if thought.items else None}
+
+@router.post("/reaction/{message_id}")
+async def add_reaction(conversation_id: str, message_id: str, user_id: str, reaction: str):
+    if reaction not in ["thumbs_up", "thumbs_down"]:
+        raise HTTPException(status_code=400, detail="Invalid reaction type")
+
+    user = honcho.apps.users.get_or_create(app_id=app.id, name=user_id)
+
+    # Check if a reaction already exists
+    existing_reaction = honcho.apps.users.sessions.metamessages.list(
```

Comment:
I don't think metamessages really make sense for this. I was thinking more just update the metadata of the message itself. That way you don't need an additional API request to get the reaction data it should come with the initial message itself. 

Think of metamessages more for storing intermediate inferences in an LLM architecture of bigger pieces of context. For something at this scale I think it might be more overhead. 

Using the metadata should also help with the loading jitter

Reply:
@VVoruganti got it! Using message metadata now.